<launch>

    <arg name="zmq_host" default="localhost" doc="Hostname or IP of the ZMQ server" />
    <arg name="zmq_port" default="5555" doc="Port of the ZMQ server" />
    <arg name="rate" default="10.0" doc="Processing rate in Hz" />
    <arg name="camera_namespace" default="cameras/cam_front" doc="Namespace of the camera (used for intrinsics)" />
    <arg name="input_image_topic" default="$(arg camera_namespace)/image_raw" doc="Input image topic to subscribe to" />
    <arg name="output_depth_topic" default="depth" />
    <arg name="output_colorized_topic" default="depth/colorized" />

    <node pkg="auv_vision" type="depth_anything_client.py" name="depth_anything_node" output="screen">

        <param name="zmq_host" value="$(arg zmq_host)" />
        <param name="zmq_port" value="$(arg zmq_port)" />
        <param name="rate" value="$(arg rate)" />
        <param name="camera_namespace" value="$(arg camera_namespace)" />

        <remap from="image_raw" to="$(arg input_image_topic)" />

        <remap from="~depth" to="$(arg output_depth_topic)" />
        <remap from="~colorized" to="$(arg output_colorized_topic)" />
    </node>
</launch>
